### PR TITLE
Out of bounds write example

### PR DESCRIPTION
**Pull request by jonmattgray**
An example of a user changing the root password by entering a username that's long enough (>12 characters) to go out of bounds of the allocated input array.
The password is replaced with the out of bounds characters e.g. if `root--------123` is inputted for a username, the password is now `123`.

Baseline:
image

CHERI:
image

### CODE REVIEW

File: src/out-of-bounds-access.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include <stdlib.h>
+
+// An example of a user changing the root password by entering a username that's long enough (>10 characters) to go out of bounds of the allocated input array.
+// The password is replaced with the out of bounds characters e.g. if `root------123` is inputted for a username, the password is now `123`.
+
+int checkInput(char input[], char toMatch[])
+{
+    input[strcspn(input, "\n")] = '\0'; // strip \n from end of fgets
+    if (strcmp(input, toMatch))
+    {
```
Comment (n3op2) [inline src/out-of-bounds-access.c:13]: Nothing against, but why this formatting?

File: src/out-of-bounds-access.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+}
+
+void runAttempt(char input[], char username[], char password[])
+{
+    printf("\nEnter username: \n");
+    fgets(input, 20, stdin);
+    if (!checkInput(input, username))
+    {
+        return;
```
Comment (n3op2) [inline src/out-of-bounds-access.c]: Basically blank return is not the best practice, normally anything that is not 0 considered to be error. So maybe converting this function into `int runAttempt(.....` and then return

Comment (n3op2) [reply src/out-of-bounds-access.c]: or a bool e.g. return true/false I don't know just being pedantic I guess. Anyway I'm not seeing as a major so approving

**Review by n3op2 — APPROVED** [review]

File: src/out-of-bounds-access.c (hunk 1 above)
Comment (jonmattgray) [reply src/out-of-bounds-access.c:13]: `{` on new line seems to be the convention (looking at other c files in the repo). My formatter did it for me automatically

File: src/out-of-bounds-access.c (hunk 2 above)
Comment (jonmattgray) [reply src/out-of-bounds-access.c]: changed to return int

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**